File: api-ventas-admin/src/main/java/com/cibertec/model/Usuario.java
// Usuario.java
package com.cibertec.model;

import java.time.LocalDateTime;
import jakarta.persistence.*;
import lombok.*;

@Data
@Entity
@Table(name = "Usuario")
@NoArgsConstructor
@AllArgsConstructor
@Builder 
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idUsuario")
    private Integer idUsuario;

    @Column(length = 100)
    private String nombreCompleto;

    @Column(length = 40, unique = true)
    private String correo;

    @ManyToOne
    @JoinColumn(name = "idRol", nullable = false)
    private Rol rol;

    @Column(length = 40)
    private String clave;

    private Boolean esActivo;

    @Column(insertable = false, updatable = false)
    private LocalDateTime fechaRegistro;
}

File: AppSistemaVentaAdmin/src/app/Interfaces/usuario.ts
export interface Rol {
  idRol: number;
  nombre: string;
  fechaRegistro: string;
}

export interface Usuario {
  idUsuario: number;
  nombreCompleto: string;
  correo: string;
  clave: string;
  esActivo: boolean;
  fechaRegistro: string;
   rol: Rol;
}
